### PR TITLE
Add support for `text-to-speech` (w/ Speecht5)

### LINKED ISSUE #59: [Feature request] Add text-to-speech with SpeechT5
**Name of the feature**
Speech to text using SpeechT5, which was recently added to Transformers.

* **Blog post**: https://huggingface.co/blog/speecht5
* **Spaces demo**: https://huggingface.co/spaces/Matthijs/speecht5-tts-demo
* **Models**: https://huggingface.co/mechanicalsea/speecht5-tts
* **Github repo**: https://github.com/microsoft/SpeechT5/
* **Paper**: https://arxiv.org/abs/2110.07205
* **Speaker embedding creation**: https://huggingface.co/mechanicalsea/speecht5-vc/blob/main/manifest/utils/prep_cmu_arctic_spkemb.py

**Reason for request**
The brower's default TTS API is quite bad if you want to create an experience that works nicely across all browsers. Firefox's voices in particular are extremely robotic. Some applications require that the voice is consistent, and of a particular style/tone/etc. SpeechT5 allows you to create 512-dim speaker embeddings so you can use an arbitrary voice style.

**Additional context**
* The model runs in realtime on the CPU (Pytorch), so with WebGPU we should easily have realtime generation on the web.
* According the the above-linked models repo, the models are 600M (T5) and 300M (Hi-Fi-GAN), but I've just tried running it locally with [the new docker integration on Hugging Face](https://twitter.com/huggingface/status/1641029818635493382) and it downloads a 585M model and a 50M model. So I'm not sure what's going on with the GAN size difference. Maybe they have quantized the GAN, but not T5? Hoping tha the T5 model can be quantized because that would move it from "reasonable" to "good" territory in terms of size. I'm assuming that it's currently in 16 bit format.

Example clip from the Spaces demo (this embedding is pretty monotone):

[tmptgsysvc8.webm](https://user-images.githubusercontent.com/1167575/228582485-74be712c-f01b-43d0-90f5-4322f1facade.webm)

### PR DESCRIPTION
**Pull request by xenova**
This PR adds text-to-speech support to Transformers.js, with speecht5. We will add support for bark and other models in future updates (and when Optimum supports those exports).

closes #59, #279, #315

## Example usage:
```js
import { pipeline } from '@xenova/transformers'

// Choose speaker embeddings
let speaker_embeddings = 'https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/speaker_embeddings.bin';

// Create pipeline (NOTE: unquantized)
let synthesizer = await pipeline('text-to-speech', 'Xenova/speecht5_tts', { quantized: false });

// Generate audio from text
let out = await synthesizer('Hello, my dog is cute', { speaker_embeddings });
// {
//   audio: Float32Array(28672) [-0.0004919943166896701, -0.00023953932395670563, ...],
//   sampling_rate: 16000
// }
```

```js
// (Optional) Write to .wav file
import fs from 'fs';
import wavefile from 'wavefile';

let wav = new wavefile.WaveFile();
wav.fromScratch(1, out.sampling_rate, '32f', out.audio);
fs.writeFileSync('out.wav', wav.toBuffer());
```

https://github.com/xenova/transformers.js/assets/26504141/0b301849-ad47-4def-b909-f6ba927c84c4

(converted to mp4 since GH doesn't allow wav)

Notes:
- If you use the quantized versions, you'll get poor results. I believe this is because the mel-spectrogram calculations require high precision. (cc @Vaibhavs10)
- ~There are minor artifacts in the output, so just need to check this (cc @fxmarty). e.g., here is the python output:~

https://github.com/xenova/transformers.js/assets/26504141/41dcc119-4783-428a-83b4-4ba39ff8e33f

### TODO:
- [x] Create server-side audio processing guide (node.js/deno)
- [x] Create in-browser audio processing guide (w/ web audio api)
- [x] Experiment with different quantization settings

### CODE REVIEW

**Comment by xenova** [comment]
My `generate_speech` function is based on [this code block](https://github.com/huggingface/optimum/pull/1404#issuecomment-1735963247) by @fxmarty, however, it also produces artifacts in the output:


https://github.com/xenova/transformers.js/assets/26504141/405117f1-2e5c-45ba-81d3-b1aee006101b

(running that code in python w/ same speaker embeddings).

My hunch is that the problem is either:
- related to dropout, or
- related to [this missing section](https://github.com/huggingface/transformers/blob/fc296f419e451d3c595a68455f18e20a3819fcd5/src/transformers/models/speecht5/modeling_speecht5.py#L2554-L2558):
```py
    # downsample encoder attention mask
    if isinstance(model.speecht5.encoder, SpeechT5EncoderWithSpeechPrenet):
        encoder_attention_mask = model.speecht5.encoder.prenet._get_feature_vector_attention_mask(
            encoder_out[0].shape[1], encoder_attention_mask
        )
```

will continue investigating.

**Comment by xenova** [comment]
@fxmarty The dropout patch fixed it! 🥳 

https://github.com/xenova/transformers.js/assets/26504141/0b301849-ad47-4def-b909-f6ba927c84c4

**Comment by xenova** [comment]
And the audio difference can be fixed by multiplying the waveform by some constant factor. I don't see any post-processing in the [`generate_speech` function in transformers](https://github.com/huggingface/transformers/blob/0a3b9d02fed3170965d359ea31f7d26651858306/src/transformers/models/speecht5/modeling_speecht5.py#L2534), so I assume the JS library I used to export (`wavefile`) and the python library I used to export (`soundfile`) have different scaling defaults for 1-channel audio. Either way, not an issue here.

#### Python

https://github.com/xenova/transformers.js/assets/26504141/95941979-9852-4ad9-9fa2-569a94337cd2


#### JavaScript
- no scaling

    https://github.com/xenova/transformers.js/assets/26504141/513973f9-223f-4365-b7b0-f96ddcafd609

- scale by sqrt(2)

    https://github.com/xenova/transformers.js/assets/26504141/003a36f8-1634-475f-8b87-78c56e52f0e7

- scale by 2

    https://github.com/xenova/transformers.js/assets/26504141/4871c76c-4595-4e8d-8f59-0d716ebc33b3

**Comment by flatsiedatsie** [comment]
I tried to use WebGPU in v3 to increase the generation speed, but got an error. Would it be difficult to add WebGPU support for TextToSpeech?
